driver-core/src/main/com/mongodb/client/model/search/SearchOperator.java

```diff
@@ -292,6 +299,96 @@ static GeoNearSearchOperator near(final Point origin, final Number pivot, final
+     * @mongodb.atlas.manual atlas-search/equals/ equals operator
+     */
+    static EqualsSearchOperator equals(final FieldSearchPath path, final boolean value) {
+        return new SearchConstructibleBsonElement("equals", new Document("path", notNull("path", path).toValue())
+                .append("value", value));
+    }
+
+    /**
+     * Returns a {@link SearchOperator} that checks whether a field matches a value you specify.
+     *
+     * @param path The indexed field to be searched.
+     * @param value The object id value to query for.
+     * @return The requested {@link SearchOperator}.
+     * @mongodb.atlas.manual atlas-search/equals/ equals operator
+     */
+    static EqualsSearchOperator equals(final FieldSearchPath path, final ObjectId value) {
+        return new SearchConstructibleBsonElement("equals", new Document("path", notNull("path", path).toValue())
+                .append("value", notNull("value", value)));
+    }
+
+    /**
+     * Returns a {@link SearchOperator} that checks whether a field matches a value you specify.
+     *
+     * @param path The indexed field to be searched.
+     * @param value The number value to query for.
+     * @return The requested {@link SearchOperator}.
+     * @mongodb.atlas.manual atlas-search/equals/ equals operator
+     */
+    static EqualsSearchOperator equals(final FieldSearchPath path, final Number value) {
+        return new SearchConstructibleBsonElement("equals", new Document("path", notNull("path", path).toValue())
+                .append("value", notNull("value", value)));
+    }
+
+    /**
+     * Returns a {@link SearchOperator} that checks whether a field matches a value you specify.
+     *
+     * @param path The indexed field to be searched.
+     * @param value The instant date value to query for.
+     * @return The requested {@link SearchOperator}.
+     * @mongodb.atlas.manual atlas-search/equals/ equals operator
+     */
+    static EqualsSearchOperator equals(final FieldSearchPath path, final Instant value) {
+        return new SearchConstructibleBsonElement("equals", new Document("path", notNull("path", path).toValue())
+                .append("value", notNull("value", value)));
+    }
+
+    /**
+     * Returns a {@link SearchOperator} that checks whether a field matches a value you specify.
+     *
+     * @param path The indexed field to be searched.
+     * @param value The string value to query for.
+     * @return The requested {@link SearchOperator}.
+     * @mongodb.atlas.manual atlas-search/equals/ equals operator
+     */
+    static EqualsSearchOperator equals(final FieldSearchPath path, final String value) {
+        return new SearchConstructibleBsonElement("equals", new Document("path", notNull("path", path).toValue())
+                .append("value", notNull("value", value)));
+    }
+
+    /**
+     * Returns a {@link SearchOperator} that checks whether a field matches a value you specify.
```

Comment:
The docs from our docs page are confusing. I think we should say, `...that searches for documents where a field matches the specified value.` in all of these